src/main/java/com/xmage/launcher/XMageLauncher.java

```diff
@@ -400,28 +340,32 @@ private static void openWebpage(String uri) {
         if (desktop != null && desktop.isSupported(Desktop.Action.BROWSE)) {
             try {
                 desktop.browse(new URI(uri));
-            } catch (URISyntaxException ex) {
-                logger.error("Error: ", ex);
-            } catch (IOException ex) {
+            } catch (URISyntaxException | IOException ex) {
                 logger.error("Error: ", ex);
             }
         }
     }
 
     private void handleClient() {
+        checkJava();
         Process p = Utilities.launchClientProcess();
-        clientConsole.setVisible(Config.isShowClientConsole());
+        clientConsole.setVisible(Config.getInstance().isShowClientConsole());
         clientConsole.start(p);
+        clientProcesses.add(p);
     }
 
     private void handleServer() {
         if (serverProcess == null) {
+            checkJava();
```

Comment:
Alright, found a bug that needs fixing before merging (currently writing up changelog for this)


Steps:
1. Boot, tell it to use system JRE
2. Download the client and server
3. Switch from using the system JRE under settings
4. Click run client/server. When prompted about Java, click "OK" instead of "USE SYSTEM JRE"

Problem:
- The bundled java does **not** get downloaded, and the server won't run. The user must click "Update" manually in order to get the java to download

Reply:
I'll add a reminder for the user + a check for that

Reply:
Rather than a reminder, can you have it auto-download When the OK button is checked. (Or When the clients/server) try to be run?

Reply:
It's not a good idea: the user should be able to explicitly decide when downloads should happen.
I'll add an error message that says something like "You need to update XMage first"

Reply:
Then you need to:
1. Change the text on the pop up since it currently says that a new download will occur.
2. Grey out the launch client/server buttons the way they are greyed out when they aren't available

Reply:
Done

Reply:
Better, but still not there.

If the user changes the java setting (system -> downloaded) or (downloaded -> system) the check for an available java should happen when they click "Done" to close the settings.

E.g. **I switch from system JRE to downloaded one in settings**
Expected behaviour: 
- "Launch" buttons are greyed out if I don't have the java installed

Current behaviour:
- Launch buttons remain enabled
- Clicking on them asks me, again, if I would like to use my system JRE or the downloaded one. (it should not ask again since I **just** changed)
- I am prompted to update
- Buttons then get greyed out.

**Further change:**
- [ ] The "Update" button should say "Install" if either XMage or the downloaded Java are not installed.

Reply:
Done, but I didn't rename the button in order to avoid confusions when messages refer to the "Update" button and users see "Install" instead

Reply:
Still not working properly:

Reply:
> Done, but I didn't rename the button in order to avoid confusions when messages refer to the "Update" button and users see "Install" instead

That's fine for now

Reply:
Still not working properly:
- [ ] Switching from system java to downloaded java still displays a notification saying that "this is the first time the client" has been run. That notification shouldn't show up for the user since they have just switched java version
- [ ] Switching immediately back to system java does not re-enable the launch buttons

Reply:
> Still not working properly:
> 
> * [x]  Switching from system java to downloaded java still displays a notification saying that "this is the first time the client" has been run. That notification shouldn't show up for the user since they have just switched java version
> * [x]  Switching immediately back to system java does not re-enable the launch buttons

Try now

Reply:
> > Still not working properly:
> > 
> > * [x]   Switching from system java to downloaded java still displays a notification saying that "this is the first time the client" has been run. That notification shouldn't show up for the user since they have just switched java version
> > * [x]   Switching immediately back to system java does not re-enable the launch buttons
> 
> Try now

That all looks good!

Reply:
> It's not a good idea: the user should be able to explicitly decide when downloads should happen.
> I'll add an error message that says something like "You need to update XMage first"

I want to come back to this since it's not sitting well with me, especially given the first run experience now. 

When the user opens Xmage for the firs time, they are told that XMage must download itself in order to play. Which I believe is the correct way to do this. The only reason they booted the launcher would be to play.

If the user also chooses to use the downloaded java (which they are prompted to choose **before** being told that xmage will download itself) then the JRE will also be downloaded automatically along with xmage.

The same should be true if the user switches from the system java to the downloaded one. When they switch to the downloaded one they are provided with a prompt telling them that there is no java installed. and asks them what do to. Letting them switch back to the installed java incase they don't want to download something is fine. But:
- [ ] Hitting okay should begin downloading the java. The user clicking okay has agreed that they want to use the downloaded java, and know that it isn't installed. There is no reason to make them click update separately.

Reply:
Ok I'll do it your way

Reply:
Try now